webview-ui/src/components/settings/RateLimitControl.tsx

```diff
@@ -0,0 +1,73 @@
+import { VSCodeCheckbox } from "@vscode/webview-ui-toolkit/react"
+import { useEffect, useState, FormEvent } from "react"
+import { useAppTranslation } from "@/i18n/TranslationContext"
+import { useDebounce } from "react-use"
+
+interface RateLimitControlProps {
+	value: number | undefined | null
+	onChange: (value: number | undefined | null) => void
+	maxValue?: number
+}
+
+export const RateLimitControl = ({ value, onChange, maxValue = 60 }: RateLimitControlProps) => {
+	const { t } = useAppTranslation()
+	const [isCustomRateLimit, setIsCustomRateLimit] = useState(value !== undefined)
+	const [inputValue, setInputValue] = useState(value)
+	useDebounce(() => onChange(inputValue), 50, [onChange, inputValue])
+	// Sync internal state with prop changes when switching profiles
+	useEffect(() => {
+		const hasCustomRateLimit = value !== undefined && value !== null
```

Comment:
This checkbox doesn't seem to be serializing correctly

https://github.com/user-attachments/assets/f48b8253-693f-4c83-8dc4-117b5d174b2b